app/app.ino

```diff
@@ -19,52 +24,27 @@ void initWifi()
     }
 }
 
-void initIoThub() 
+void setup()
 {
-    if (platform_init() != 0)
-    {
-        LogInfo("Failed to initialize the platform.");
-        while (1);
-    }
-
-    iotHubClientHandle = IoTHubClient_LL_CreateFromConnectionString(CONNECTION_STRING, MQTT_Protocol);
-    if (iotHubClientHandle == NULL)
-    {
-        LogInfo("Failed on IoTHubClient_CreateFromConnectionString");
-        while (1);
-    }
-
-    if (IoTHubClient_LL_SetOption(iotHubClientHandle, "TrustedCerts", certificates) != IOTHUB_CLIENT_OK)
-    {
-        LogInfo("Failed to set option \"TrustedCerts\"");
-        while (1);
-    }
-
-    if (IoTHubClient_LL_SetDeviceTwinCallback(iotHubClientHandle, twinCallback, NULL) != IOTHUB_CLIENT_OK)
+    hasWifi = false;
+    initWifi();
+    if (!hasWifi)
     {
-        LogInfo("Failed on IoTHubClient_LL_SetDeviceTwinCallback");
-        while (1);
+        LogInfo("Please make sure the wifi connected!");
+        return;
     }
-}
 
-void setup()
-{
     Serial.begin(9600);
-
-    initWifi();
-    initSensor();
-    initIoThub();
+    sensorInit();
+    iothubInit();
 }
 
 void loop()
 {
-    Serial.print("Humidity: ");
-    Serial.println(readHumidity());
-
-    Serial.print("Temperature: ");
-    Serial.println(readTemperature());
-
-    IoTHubClient_LL_DoWork(iotHubClientHandle);
-
-    delay(interval);
+    char messagePayload[MESSAGE_MAX_LEN];
+    readMessage(messageCount, messagePayload);
+    iothubSendMessage((const unsigned char *)messagePayload);
+    messageCount++;
+    iothubLoop();
+    delay(getInterval());

```

Comment:
Align with Happy Path, change here with delay(10), and put `delay(getInterval());` into ` if (!messagePending && messageSending)`